### PR TITLE
Bump github.com/cpuguy83/go-md2man dependency to v2.0.0

### PR DESCRIPTION
**Pull request by munnerz**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Bumps the dependency on `github.com/cpuguy83/go-md2man` to v2.0.0, as I'm running into some weird issues around this dependency when running `update-vendor-licenses.sh` in the cert-manager repo, and want to see if it works here.

Also just as a general cleanup task to remove the dependency on the v1 module here 😄 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/cc @liggitt

### CODE REVIEW

File: LICENSES/vendor/go.opencensus.io/LICENSE [hunk 1]
```diff
@@ -0,0 +1,205 @@
+= vendor/go.opencensus.io licensed under: =
```
[the remark this answers is not in the document]
Comment (liggitt) [reply LICENSES/vendor/go.opencensus.io/LICENSE:1]: well that's disconcerting

[the remark this answers is not in the document]
Comment (liggitt) [reply LICENSES/vendor/go.opencensus.io/LICENSE:1]: drop from this PR, opened https://github.com/kubernetes/kubernetes/issues/92728 to track this and fixed in https://github.com/kubernetes/kubernetes/pull/92729

[the remark this answers is not in the document]
Comment (munnerz) [reply LICENSES/vendor/go.opencensus.io/LICENSE]: done

**Comment by munnerz** [comment]
`go-md2man` is only used by `cmd/genman`, which generates cli man pages, so...

/sig cli

**Pull request merged**